packages/models/src/crypto-assets/crypto-asset-info.model.ts

```diff
@@ -1,68 +1,114 @@
 import { InscriptionMimeType } from './bitcoin/inscription.model';
 
+export const CryptoAssetCategories = {
+  fungible: 'fungible',
+  collectible: 'collectible',
+} as const;
+export const CryptoAssetTypes = {
+  nativeBtc: 'nativeBtc',
+  nativeStx: 'nativeStx',
+  sip10: 'sip10',
+  brc20: 'brc20',
+  src20: 'src20',
+  stx20: 'stx20',
+  rune: 'rune',
+  stamp: 'stamp',
+  sip9: 'sip9',
+  inscription: 'inscription',
+} as const;
+export type CryptoAssetCategory = keyof typeof CryptoAssetCategories;
+export type CryptoAssetType = keyof typeof CryptoAssetTypes;
+
 export interface BaseCryptoAssetInfo {
+  readonly assetCategory: CryptoAssetCategory;
+  readonly assetType: CryptoAssetType;
+  readonly name: string;
+}
+
+// Fungible asset types
+interface BaseFungibleCryptoAssetInfo extends BaseCryptoAssetInfo {
+  readonly assetCategory: 'fungible';
+  readonly symbol: string;
   readonly decimals: number;
   readonly hasMemo: boolean;
 }
-
-export interface BtcCryptoAssetInfo extends BaseCryptoAssetInfo {
+export interface BtcCryptoAssetInfo extends BaseFungibleCryptoAssetInfo {
+  readonly assetType: 'nativeBtc';
   readonly name: 'bitcoin';
   readonly symbol: 'BTC';
 }
-
-export interface StxCryptoAssetInfo extends BaseCryptoAssetInfo {
+export interface StxCryptoAssetInfo extends BaseFungibleCryptoAssetInfo {
+  readonly assetType: 'nativeStx';
   readonly name: 'stacks';
   readonly symbol: 'STX';
 }
-
-// Bitcoin
-export interface Brc20CryptoAssetInfo extends BaseCryptoAssetInfo {
+export interface Brc20CryptoAssetInfo extends BaseFungibleCryptoAssetInfo {
+  readonly assetType: 'brc20';
   readonly name: 'brc-20';
   readonly symbol: string;
 }
-
-export interface InscriptionCryptoAssetInfo {
+export interface Src20CryptoAssetInfo extends BaseFungibleCryptoAssetInfo {
+  readonly assetType: 'src20';
+  readonly name: 'src-20';
```

Comment:
Do we need both `assetType` and `name`? I believe they might be being used in the same way?

Reply:
`assetType` is stictly used as a discriminator, whereas `name` was already being used differently in different contexts and cant be used for this purpose. Most assets have a similar "asset type" pre-assigned in the `name` field, but others type this field as a `string` and fill it differently (e.g. in sip-10 tokens `name` is populated with "ALEX token", etc..)

Reply:
Following from other comment: these interfaces describe asset types, so the `name` refers to the `assetType`, which is why this feels like duplication. 

If ALEX token is being passed to `name` in these asset then therein lies the problem. Do we need a larger refactor here to fix this. That'd be a better solution than adding a new type to deal with existing tech debt. Maybe this is something that'd be good to discuss live?

Reply:
@kyranjamie I agree we can & should refactor `name` out of assets that duplicate `assetType`. I didn't want to do that here to keep scope reasonable (will take quite some work to support in `extension`) and to get feedback first on the changes.

The model rework does go beyond just adding a discriminable `assetType` field, and establishes more collective types and a clearer hierarchy. Utilizing discrimination and shared properties can be useful in many contexts.

Reply:
I know that `name` was setup to be a discriminating union type, so if it has been implemented in correctly somewhere, then it should prob be fixed rather than adding to the model? Also, we know these are assets, so why is it nec to include `asset` in the prop?

Reply:
My comment here is just meant to say, we should make sure to make an issue to refactor these so these props aren't duplicated. I am fine if you change them in a way that makes more sense with your work here.

Reply:
@fbwoolf The `asset` prefix is a bit redundant, but I've found that generic field names like `type` tend to lose context and can benefit from extra clarity. For example child, interfaces inherit all parent fields, so if we add another "type" somewhere down the line, like on an NFT asset, the benefit of the prefix becomes more obvious:
```
interface Sip9CryptoAssetInfo extends BaseCollectibleCryptoAssetInfo {
  assetType: 'sip9';
  collectionType: 'generative' | 'series';
  contractType: 'basic' | 'wrapped';
  mintType: ...
}
```

Totally agree on fixing `name`. It seems to be used mostly as like a display name in `extension` but it's hard to tell exactly where and how it's used. Definitely deserves its own issue / PR.

Reply:
Yeah, good point, I remembered `mimeType` after I made the comment ...makes sense to keep it. 👍